api/lightning_utils.py

```diff
@@ -0,0 +1,189 @@
+import requests, json
+import models as m
+import time
+from extensions import db
+from flask import current_app as app
+
+auth_header = ''
+lndhub_url = ''
+
+class LightningInvoiceUtil:
+    def __init__(self):
+        self.lndhub_url = app.config['LNDHUB_URL']
+        app.logger.info(f"init - LNDHUB_URL={self.lndhub_url}")
+        self.auth_header = self.get_login_token()
+
+    def get_login_token(self):
+        app.logger.info(f"get_login_token - Trying login to LndHub API ({self.lndhub_url})...")
+
+        payload = {
+            'login': app.config['LNDHUB_USER'],
+            'password': app.config['LNDHUB_PASSWORD']
+        }
+
+        r = requests.post(
+            self.lndhub_url + '/auth',
+            data=payload
+        )
+
+        json_res = r.json()
+        app.logger.debug(f"get_login_token - json_res = ({json_res})...")
+
+        if 'error' in json_res:
+            app.logger.error(f"get_login_token - {json_res['message']}")
+            time.sleep(60)
```

Comment:
The decision to sleep on error should be left to the caller, not to a method that is just supposed to get the token or not...